### PR TITLE
Replace delimited_payload_filter by delimited_payload

### LINKED ISSUE #21978: Rename `delimiter_payload_filter` to `delimiter_payload`.
This pre-built token filter is inconsistent with other pre-built token filters since it includes `filter` in its name while the other ones don't.

### PR DESCRIPTION
**Pull request by liketic**
Closes #21978 . Any comments are appreciated. 😄

### CODE REVIEW

**Comment by cbuescher** [comment]
I believe the issue this is trying to fix is https://github.com/elastic/elasticsearch/issues/21978.
@liketic The PR seems a bit out of date, would you mind rebasing or merging master to this branch so it can be tested?

**Comment by liketic** [comment]
@cbuescher Thanks for your attention! Yes, my mistake. I rebased the pr, please review and test. Thanks a lot! 👍

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@elasticmachine test this please

File: modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java [hunk 1]
```diff
@@ -191,10 +193,15 @@
                 input -> new CommonGramsFilter(input, CharArraySet.EMPTY_SET)));
         filters.add(PreConfiguredTokenFilter.singleton("czech_stem", false, CzechStemFilter::new));
         filters.add(PreConfiguredTokenFilter.singleton("decimal_digit", true, DecimalDigitFilter::new));
+        // TODO deprecate delimited_payload_filter
```
Comment (cbuescher) [inline modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java]: I think the deprecation also needs to happen in this PR. Using the old name, either in the settings when creating an index or when using it in an already existing index should trigger a deprecation warning and an entry in the logs.

**Review by cbuescher — CHANGES_REQUESTED** [review]

**Comment by liketic** [comment]
@cbuescher Thanks for your help!  Could you guide me where to add that log? Thanks in advance!

**Comment by cbuescher** [comment]
@liketic I have to admiz I'm not entirely sure where the best place would be to add that kind deprecation logging. I would need to do some digging as well, but one idea to look at might be the AnalysisRegistry itself, where I think all calls to obtain an instance of a named token filter need to go though "getTokenFilterProvider()". Not sure if this is the most elegant place to put the deprecation though.

File: modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java (hunk 1 above)
Comment (liketic) [reply modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java]: I pushed 7e7c54dce07cb806477e34ff81f2b46ea87edfb3

**Comment by liketic** [comment]
@cbuescher Thanks! I'm trying to add deprecation log in ```AnalysisRegistry```, please review again.

**Comment by jpountz** [comment]
I think a better way to do this would be to register `delimited_payload_filter` with a wrapper of the analysis factory of `delimited_payload` that emits a deprecation log when the version is >= 7.0. This way old indices that use `delimited_payload_filter` would not pollute the deprecation log since they would still be using the deprecated filter (which they cannot change).

**Comment by liketic** [comment]
@jpountz I tried to add a wrapper class. I'm not sure it's correct. Could  you please review it?

File: modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/DelimitedPayloadTokenFilterFactoryWrapper.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+package org.elasticsearch.analysis.common;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.common.logging.DeprecationLogger;
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.index.IndexSettings;
+
+public class DelimitedPayloadTokenFilterFactoryWrapper extends DelimitedPayloadTokenFilterFactory {
```
Comment (jpountz) [inline modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/DelimitedPayloadTokenFilterFactoryWrapper.java]: Maybe rename it to something like `LegacyDelimitedPayloadTokenFilterFactory` but other than that, this is indeed the approach I was thinking of.

**Review by jpountz — APPROVED** [review]
I left a comment about naming but things look good to me otherwise.

File: modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/DelimitedPayloadTokenFilterFactoryWrapper.java (hunk 2 above)
Comment (liketic) [reply modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/DelimitedPayloadTokenFilterFactoryWrapper.java]: Thanks！

**Comment by liketic** [comment]
Hi @cbuescher Could you please review again? Thanks in advance.

File: docs/reference/migration/migrate_7_0/analysis.asciidoc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[[breaking_70_analysis_changes]]
+=== Analysis changes
+
+==== The `delimited_payload_filter` is deprecated
+
+The `delimited_payload_filter` is deprecated and should be replaced by `delimited_payload`.
```
Comment (cbuescher) [inline docs/reference/migration/migrate_7_0/analysis.asciidoc]: Maybe we should make it clearer that the filter is just renamed, and its only the old name that is deprecated, something like "The `delimited_payload_filter` is renamed to `delimited_payload`, the old name is deprecated and will be removed at some point, so it should be replaces by `delimited_payload`" or slightly different.

**Review by cbuescher — APPROVED** [review]
@liketic thanks, this looks also good to me. I left a tiny comment regarding the migration note but other than that I agree this is ready. I will kick off the remaining CI tests once this small change has been adressed and will continue to merge after CI is green.

**Comment by liketic** [comment]
@cbuescher Thanks for your help. I pushed d3e6cfbba7972f0ea0a3145b4768ce8711ff0351 . 👍

**Comment by cbuescher** [comment]
@liketic thanks a lot for the last change
@elasticmachine test this please

**Comment by cbuescher** [comment]
@liketic seems like there are still some checkstyle conventions that CI is complaining about now, sorry about that:
```
LegacyDelimitedPayloadTokenFilterFactory.java:30: Line is longer than 140 characters
```
btw. you can run `gradle precommit` (this skips all the long-running tests) either in the root folder or the affected module to check for these kind of errors locally, but I also keep forgetting that. Can you change that line and then run these checks locally once to make sure there are no other style errors?

**Comment by liketic** [comment]
@cbuescher My fault. Please test again. Thanks!

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@liketic thanks, the checkstyle etc... passes now, the current errors seem unrelated to this PR but I'd still like to try to get a clean run. Would you be able to merge in master once more so we get a current state there?

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@liketic sorry this took a while longer, but thanks for this PR. I just merged it to master.
@jpountz maybe we can backport this to one of the coming 6.x versions? If we deprecate in, say 6.2., like in this PR, would you be okay to remove in 7.0 or should we leave this as is, wdyt?

**Comment by lcawl** [comment]
I added the following in elastic/elasticsearch/migration/migrate_7_0.asciidoc to address documentation build errors:

> include::migrate_7_0/analysis.asciidoc[]

**Comment by cbuescher** [comment]
@lcawl thanks for fixing this and sorry I didn't catch this in the review.

**Comment by lcawl** [comment]
No problem, it was a quick fix!

**Comment by clintongormley** [comment]
> @jpountz maybe we can backport this to one of the coming 6.x versions? If we deprecate in, say 6.2., like in this PR, would you be okay to remove in 7.0 or should we leave this as is, wdyt?

I think we should deprecate in 6.x and remove in 7.0, but still allow the old form in 7.0 for indices created in 6.x.

Also, we should add a check for this to the x-pack migration assistant.

**Comment by cbuescher** [comment]
@clintongormley I opened #27704 for the final removal in 7.0 and adding checks to the migration assistant.

**Pull request merged**